app/controllers/analytics_controller.rb

```diff
@@ -0,0 +1,24 @@
+class AnalyticsController < ApplicationController
+  before_action :custom_method, :only => [:show, :index]
+
+  def show
+  end
+
+  def index
+    @last_days = 30
+    @full_width_content = true
+    @visits = Visit.where(started_at: @last_days.days.ago..Time.now)
+    @views = Ahoy::Event.where(name: '$view' )
+  end
+
+private
```

Comment:
there is no particular reason to call it custom_method. I can change the name to something more appropriate.